### PR TITLE
Sign messages after hash verification

### PR DESCRIPTION
**Pull request by garious**
#### Problem

Can't use Ledger Wallet to sign Solana messages.

#### Proposed Changes

Allow it to sign messages after the user reviews the message's SHA256 hash.

### CODE REVIEW

**Comment by aeyakovenko** [comment]
how do I know the hash is valid?  computer can be compromised and display a hash to some random transaction that I verify on the ledger to be the same invalid hash.  User needs to actually validate the destination and the instruction.

**Comment by garious** [comment]
@aeyakovenko, you don't. Verifying the hash is the closest thing to blind signing without blindly signing.  Baby steps.  With this functionality in place, @CriesofCarrots can move forward with CLI changes, while I start parsing binary messages and displaying its pieces for the user to verify.

File: src/utils.h [hunk 1]
```diff
@@ -51,3 +54,16 @@ void sendResponse(uint8_t tx, bool approve);
...
 
 #endif
+
+#ifdef TEST
+#include <stdio.h>
+#define THROW(code) do { printf("error: %d", code); } while (0)
+#define PRINTF(msg, arg) printf(msg, arg)
+#define PIC(code) code
+//#define TARGET_NANOS 1
+#define TARGET_BLUE 1
```
Comment (CriesofCarrots) [inline src/utils.h]: Why Blue?

Comment (garious) [reply src/utils.h]: copypasta

**Comment by aeyakovenko** [comment]
@garious @jstarry 

We could use wallet.solana.com as a trusted display as well.  Basically decode and show the hash and the transaction details in our hosted wallet, confirm the hash on the ledger.

**Comment by garious** [comment]
@aeyakovenko, this conversation will surely be lost. Can you create a new ticket to discuss?

**Pull request merged**